src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,250 @@
+  double L = 0;
+  double H = 0;
+  arma::mat K;
+
+  if (kernelFunction == "linear")
+  {
+    K = data.t() * data;
+  }
+
+  else if (kernelFunction == "gaussian")
+  {
+    arma::vec X2 = arma::sum(arma::pow(data, 2), 2);
+    K = X2 + (X2.t() - 2 * data * data.t());
+  }
+  else
+  {
+    K = arma::zeros(data.n_rows, data.n_cols);
+  }
+
+  while(count < max_iter)
+  {
+    size_t num_changes_alphas = 0;
+    for(size_t i = 0; i < data.n_cols; i++)
+    {
+      E(i) = b + arma::sum(alpha % (labels.t() % K.col(i))) - labels(i);
+      if ((labels(i) * E(i) < -tol && alpha(i) < C) 
+            || (labels(i) * E(i) > tol && alpha(i) > 0))
+      {
+        size_t j = rand() % data.n_cols;
+        while (j == i)
+        {
+          j = rand() % data.n_cols;
+        }
+
+        E(j) = b + arma::sum(alpha % (labels.t() % K.col(j))) - labels(j);
+        double alpha_j_old = alpha(j);
+        double alpha_i_old = alpha(i);
+        if(labels(i) == labels(j))
+        {
+          L = std::max(0.0, alpha(i) + alpha(j) - C);
+          H = std::min(C, alpha(i) + alpha(j));
+        }
+        else
+        {
+          L = std::max(0.0, alpha(j) - alpha(i));
+          H = std::min(C, C + alpha(j) - alpha(i)); 
+        }
+
+        if(L == H)
+          continue;
+
+        eta = 2 * K(i, j) - K(i, i) - K(j, j);
+
+        if(eta >= 0)
+          continue;
+
+        alpha(j) = alpha(j) - (labels(i) * (E(i) - E(j))) / eta;
+
+        alpha(j) = std::min(H, alpha(j));
+        alpha(j) = std::max(L, alpha(j));
+
+        if(abs(alpha(j) - alpha_j_old) < tol)
+        {
+          alpha(j) = alpha_j_old;
+          continue;
+        }
+
+        alpha(i) = alpha(i) + labels(i) * labels(j) * (alpha_j_old - alpha(j));
+
+        double b1 = b - E(i)
+                    - labels(i) * (alpha(i) - alpha_i_old) * K(i, j)
+                    - labels(j) * (alpha(j) - alpha_i_old) * K(i, j);
+
+        double b2 = b - E(j)
+                    - labels(i) * (alpha(i) - alpha_i_old) * K(i, j)
+                    - labels(j) * (alpha(j) - alpha_i_old) * K(j, j);
+
+        if (0 < alpha(i) && alpha(j) < C)
+        {
+          b = b1;
+        }
+        else if (0 < alpha(j) && alpha(j)< C)
+        {
+          b = b2;
+        }
+        else
+        {
+          b = (b1 + b2) / 2;
+        }
+        num_changes_alphas = num_changes_alphas + 1;
+      }
+    }
+    if (num_changes_alphas == 0)
+      count = count + 1;
+    else
+      count = 0;
+  }
+  w = ((alpha.t() % labels) * data.t());
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels) const
+{
+  arma::mat scores;
+  Classify(data, labels, scores);
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels,
+    arma::mat& scores) const
+{
+  Classify(data, scores);
+
```

Comment:
But when I print `scores` here it turns out to be zero for all.
Is it because of 
```
  if (kernelFunction == "linear")
  {
    scores = data.t() * w.t() + b;
    scores = scores.t();
  }
```
Or not I don't know.

Reply:
Can you print `b` and `w` here?

Reply:
Added all changes and printed the value of `b` and `w` . So, `b = nan` and and `w` is very small value I think there is something wrong in the implementation and sorry for delay.